packages/schemas/src/foundations/jsonb-types.ts

```diff
@@ -87,23 +90,51 @@ export enum BrandingStyle {
   Logo_Slogan = 'Logo_Slogan',
 }
 
-export const brandingGuard = z.object({
-  primaryColor: z.string(),
-  backgroundColor: z.string(),
-  darkMode: z.boolean(),
-  darkPrimaryColor: z.string(),
-  darkBackgroundColor: z.string(),
-  style: z.nativeEnum(BrandingStyle),
-  logoUrl: z.string().optional(),
-  slogan: z.string().optional(),
-});
+export const brandingGuard = z
+  .object({
+    primaryColor: z
+      .string()
+      .nonempty()
+      .refine((color) => isValidHexColor(color), invalidHexColorMessage),
+    backgroundColor: z
+      .string()
+      .nonempty()
+      .refine((color) => isValidHexColor(color), invalidHexColorMessage),
+    darkMode: z.boolean(),
+    darkPrimaryColor: z
+      .string()
+      .nonempty()
+      .refine((color) => isValidHexColor(color), invalidHexColorMessage),
+    darkBackgroundColor: z
+      .string()
+      .nonempty()
+      .refine((color) => isValidHexColor(color), invalidHexColorMessage),
+    style: z.nativeEnum(BrandingStyle),
+    logoUrl: z.string().url(),
+    slogan: z.string().nonempty().optional(),
+  })
+  .refine((branding) => {
+    if (branding.style === BrandingStyle.Logo_Slogan) {
+      return branding.slogan && branding.slogan.trim() !== '';
+    }
+
+    return true;
+  }, 'Empty branding slogan, please add it if used a UI style containing the slogan.');
 
 export type Branding = z.infer<typeof brandingGuard>;
 
-export const termsOfUseGuard = z.object({
-  enabled: z.boolean(),
-  contentUrl: z.string().optional(),
-});
+export const termsOfUseGuard = z
+  .object({
+    enabled: z.boolean(),
+    contentUrl: z.string().url().optional(),
+  })
+  .refine((termsOfUse) => {
+    if (termsOfUse.enabled) {
+      return termsOfUse.contentUrl;
```

Comment:
why return termsOfUse.contentUrl; here ?

Reply:
It's just a boolean check.
It's a little confusing.